docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,966 @@
+  DPU0            N/A               -1         Online              up       N/A
+  DPU1            N/A               -1         Online              up       N/A
+  DPU2            N/A               -1         Online              up       N/A
+  DPU3            N/A               -1         Online              up       N/A
+  DPU4            N/A               -1         Online              up       N/A
+  DPU5            N/A               -1         Online              up       N/A
+  DPU6            N/A               -1         Online              up       N/A
+  DPU7            N/A               -1         Online              up       N/A
+```
+
+#### Pass/Fail Criteria 
+ *  Verify number of DPUs from api and number of DPUs shown in the cli output.
+
+
+## Objectives of API Test Cases
+
+|    | **Test Case**   | **Intention**                              | **Comments** |
+| ---------- | ---------- | ---------------------------------------- | ---------- |
+| 1.1 | Check SmartSwitch specific ChassisClass APIs      | To verify the newly implemented SmartSwitch specific ChassisClass APIs | |
+| 1.2 | Check modified ChassisClass APIs for SmartSwitch       |  To verify the existing ChassisClass APIs that undergo minor changes with the addition of SmartSwitch| |
+| 1.3 | Check DpuModule APIs for SmartSwitch       |  To verify the newly implemented  DpuModule APIs for SmartSwitch| |
+| 1.4 | Check modified ModuleClass APIs for SmartSwitch       |  To verify the existing ModuleClass APIs that undergo minor changes with the addition of SmartSwitch| 
+
+## API Test Cases 
+
+### 1.1 Check SmartSwitch specific ChassisClass APIs
+
+#### Steps
+ * Execute the following APIs on SmartSwitch
+ * get_dpu_id(self, name):
+    * Provide name (Example: DPU0 - Get it from platform.json file)
+    * This API should return an integer from 1-8 (check it against platform.json)
+ * is_smartswitch(self):
+    * This API should return True
+ * get_module_dpu_data_port(self, index):
+    * It will return a dict as shown below.
+
+
+#### Verify in
+ * Switch
+
+#### Sample Output
+```
+On Switch:
+    get_dpu_id(self, DPU3)
+    Output: 4
+    is_smartswitch(self):
+    Output: True
+    get_module_dpu_data_port(self, DPU0):
+    Output: {
+                "interface": {"Ethernet224": "Ethernet0"}
+            }
+```
+#### Pass/Fail Criteria
+ *  The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output".
+
+
+### 1.2 Check modified ChassisClass APIs for SmartSwitch
+
+#### Steps
+ * is_modular_chassis(self):
+    * Should return False
+ * get_num_modules(self):
+    * Should return number of DPUs
+ * get_module(self, index):
+    * Make sure for each index this API returns an object and has some content and not None
+    * Check that the object's class is inherited from the ModuleBase class
+ * get_all_modules(self):
+    * This should return a list of items
```

Comment:
can we give some more details as to what list of platform items will be displayed here. Same goes for the 'dict' info in above API

Reply:
get_all_modules returns list of pointers to the object. 

get_state_info -> sample output has been added.